src/compiler/marks.ts

```diff
@@ -357,8 +357,53 @@ export namespace bar {
   }
 
   export function labels(model: Model) {
-    // TODO(#64): fill this method
-    return undefined;
+    // TODO(#64):
+    // √ horizontal support
+    // - vertical support
+    // - color support
+    // - container height/stretching consideration
+    // - consider defaults for offset
+
+    var l: any = {};
+    const orient = model.marksConfig('orient');
+
+    l.align = { value: (orient !== 'horizontal') ? "center" : "left" };
+
+    l.text = { field: model.has(LABEL) ? model.field(LABEL) : ((orient !== 'horizontal') ? model.field(Y) : model.field(X)) };
```

Comment:
right, I meant to try to pick up on whether or not the VL explicitly assigns a __field__ (see below) to the label, but you're right the way I wrote it is redundant. 

     "encoding": {
                "x": {"type": "quantitative", "field": "amount"},
                "y": {"type": "ordinal", "field": "category"},
                "label": {
                  "field": "amount",
                  "type": "quantitative"
                }
            }